bjam is stripped by default, this causes QA warning while stripping it
from do_populate_sysroot():

  WARNING: File '.../tmp/sysroots/x86_64-linux/usr/bin/bjam' \
    from bjam-native was already stripped, \
    this will prevent future debugging!

The JAM scripts allow to build unstripped version with '--debug'. Just
build and install the bjam.debug to stop bjam from being stripped in
compile step.

(From OE-Core rev: e7147de9f28925b1bb5df39d9c0848dd7957328c)

---
 bootstrap.sh                    | 1 +
 tools/build/src/engine/build.sh | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/bootstrap.sh b/bootstrap.sh
index 2723f28..4a024ea 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -228,6 +228,7 @@ if test "x$BJAM" = x; then
   echo "tools/build/v2/engine/$arch/b2"
   cp "$BJAM" .
   cp "$my_dir/tools/build/v2/engine/$arch/bjam" .
+  cp "$my_dir/tools/build/v2/engine/${arch}.debug/bjam" bjam.debug
 
 fi
 
diff --git a/tools/build/v2/engine/build.sh b/tools/build/v2/engine/build.sh
index 470ea3c..78cbb1f 100755
--- a/tools/build/v2/engine/build.sh
+++ b/tools/build/v2/engine/build.sh
@@ -299,5 +299,5 @@ if test -x "./bootstrap/jam0" ; then
     if test "${BJAM_UPDATE}" != "update" ; then
         echo_run ./bootstrap/jam0 -f build.jam --toolset=$BOOST_JAM_TOOLSET "--toolset-root=$BOOST_JAM_TOOLSET_ROOT" "$@" clean
     fi
-    echo_run ./bootstrap/jam0 -f build.jam --toolset=$BOOST_JAM_TOOLSET "--toolset-root=$BOOST_JAM_TOOLSET_ROOT" "$@"
+    echo_run ./bootstrap/jam0 -f build.jam --toolset=$BOOST_JAM_TOOLSET "--toolset-root=$BOOST_JAM_TOOLSET_ROOT" "$@" --debug
 fi
-- 
1.9.1
